packages/oid4vci/src/formats/proof-type/jwt/v-jwt-proof-type.ts

```diff
@@ -22,7 +22,10 @@ export const vCredentialRequestJwtProofTypeHeader = v.pipe(
     ({ kid, jwk }) => jwk === undefined || kid === undefined,
     `Both 'jwk' and 'kid' are defined. Only one is allowed`
   ),
-  v.check(({ trust_chain, kid }) => !trust_chain || !kid, `When 'trust_chain' is provided, 'kid' is required`)
+  v.check(
+    ({ trust_chain, kid }) => !trust_chain || kid !== undefined,
```

Comment:
Curious did you run into an issue?

Reply:
Yeah I did some work on it for supporting this property. I didn't push it with it but I had a array and a kid defined. But I still got this error so I found out that the validation is was not correct.